.gitignore

```diff
@@ -102,3 +102,6 @@ dist
 
 # TernJS port file
 .tern-port
+
+# Vim swap files
+*.swp
```

Comment:
This is fine, but you can configure vim to use a separate directory for swp files :) https://vim.fandom.com/wiki/Remove_swap_and_backup_files_from_your_working_directory